src/bip39.c

```diff
@@ -680,30 +766,91 @@ int dogecoin_generate_mnemonic (const char* entropy_size, const char* language,
 }
 
 /**
- * @brief This function derives the seed from the mnemonic
- * @param mnemonic The mnemonic code words
- * @param passphrase The passphrase (optional)
- * @param seed The 512-bit seed
+ * @brief This function derives the seed from the mnemonic.
+ *
+ * @param mnemonic The mnemonic code words.
+ * @param passphrase The passphrase (optional).
+ * @param seed The 512-bit seed.
  *
  * @return 0 (success), -1 (fail)
 */
-int dogecoin_seed_from_mnemonic (const char* mnemonic, const char* passphrase, uint8_t seed[64])
+int dogecoin_seed_from_mnemonic (const char* mnemonic, const char* passphrase, SEED seed)
 {
-    /* get seed if not null */
-    if (seed != NULL) {
+    /* Check if the input arguments are valid */
+    if (!mnemonic || !seed) {
+        fprintf(stderr, "ERROR: invalid input arguments\n");
+        return -1;
+    }
 
-        /* set passphrase to empty string if null */
-        if (passphrase == NULL) {
-            passphrase = "";
-        }
+    /* set passphrase to empty string if null */
+    if (passphrase == NULL) {
+        passphrase = "";
+    }
+
+    /* get random binary seed */
+    if (get_root_seed(mnemonic, passphrase, seed) == -1) {
+        fprintf(stderr, "ERROR: Failed to get root seed\n");
+        return -1;
+    }
+
+    return 0;
+}
 
-        /* get random binary seed */
-        if (get_root_seed(mnemonic, passphrase, seed) == -1) {
-            fprintf(stderr, "ERROR: Failed to get root seed\n");
+/**
+ * @brief This function generates a random English mnemonic phrase.
+ *
+ * @param mnemonic The mnemonic code words.
+ *
+ * @return 0 (success), -1 (fail)
+*/
+int generateRandomEnglishMnemonic (MNEMONIC mnemonic) {
+
+    /* generate an English mnemonic without random entropy */
+    return generateEnglishMnemonic (NULL, mnemonic);
+}
+
+/**
+ * @brief This function gnerates an English mnemonic phrase from hex entropy.
+ *
+ * @param mnemonic The mnemonic code words.
+ * @param entropy The hex string of entropy.
+ *
+ * @return 0 (success), -1 (fail)
+*/
+int generateEnglishMnemonic (const HEX_ENTROPY entropy, MNEMONIC mnemonic) {
+
+    /* Initialize variables */
+    char* size = "256";     /* default bits of entropy (256) */
+    char* lang = "eng";     /* default english (eng) */
+    char* space = " ";      /* default utf8 ( ) */
+    char* words = 0;        /* default no custom words (NULL) */
```

Comment:
Could these four be `const char*`, just to make it clear that these are not to be written to